web-ui/src/context/AppContext.js

```diff
@@ -55,26 +63,65 @@ const AppContextProvider = (props) => {
     updateUserProfile();
   }, []);
 
+  const date = (months, prevCheckinDate) => {
+    let newDate = prevCheckinDate ? new Date(...prevCheckinDate) : new Date();
+    newDate.setMonth(newDate.getMonth() + months);
+    const year = newDate.getFullYear();
+    const month = newDate.getMonth() + 1;
+    const day = newDate.getDate();
+    const monthArray = [year, month, day];
+    return monthArray;
+  };
+
   useEffect(() => {
     async function getCheckins() {
       if (id) {
-        let res = await getCheckinByMemberId(id);
+        const res = await getCheckinByMemberId(id);
         let data =
           res.payload &&
           res.payload.data &&
           res.payload.status === 200 &&
           !res.error
             ? res.payload.data
             : null;
-        if (data) {
-          dispatch({ type: UPDATE_CHECKINS, payload: data });
+        if (data && data.length > 0) {
+          const allComplete = data.every(
+            (checkin) => checkin.completed === true
+          );
+          if (allComplete) {
+            const prevCheckinDate = data[data.length - 1].checkInDate;
+            const res = await createCheckin({
+              teamMemberId: id,
+              pdlId: pdlId,
+              checkInDate: date(3, prevCheckinDate),
```

Comment:
What if the previous checkin date was 6 months ago? I'd say if it's more than 3 months ago, schedule it for a month out like we do if there is no check-in.